gst_plugins/python/frame_tag_filter.py

```diff
@@ -0,0 +1,233 @@
+        self.tag: Optional[str] = None
+
+        self.sink_pad: Gst.Pad = Gst.Pad.new_from_template(SINK_PAD_TEMPLATE, 'sink')
+        self.src_pad_tagged: Gst.Pad = Gst.Pad.new_from_template(
+            SRC_TAGGED_PAD_TEMPLATE, 'src_tagged'
+        )
+        self.src_pad_not_tagged: Gst.Pad = Gst.Pad.new_from_template(
+            SRC_NOT_TAGGED_PAD_TEMPLATE, 'src_not_tagged'
+        )
+        self.last_tagged: Optional[bool] = None
+
+        self.add_pad(self.sink_pad)
+        self.add_pad(self.src_pad_tagged)
+        self.add_pad(self.src_pad_not_tagged)
+
+        self.sink_pad.set_chain_function_full(self.handle_buffer)
+        self.sink_pad.add_probe(
+            Gst.PadProbeType.EVENT_DOWNSTREAM,
+            on_pad_event,
+            {Gst.EventType.CAPS: self.on_caps},
+        )
+
+    def do_state_changed(self, old: Gst.State, new: Gst.State, pending: Gst.State):
+        """Validate properties when the element is started."""
+
+        init_state = [Gst.State.NULL, Gst.State.READY]
+        if old in init_state and new not in init_state:
+            try:
+                required_property(self.source_id, 'source-id')
+                required_property(self.tag, 'tag')
+            except RequiredPropertyError as exc:
+                self.logger.exception('Failed to start element: %s', exc, exc_info=True)
+                frame = inspect.currentframe()
+                propagate_gst_setting_error(self, frame, __file__, text=exc.args[0])
+
+    def do_get_property(self, prop):
+        """Get property callback."""
+
+        if prop.name == 'tag':
+            return self.tag
+        if prop.name == 'source-id':
+            return self.source_id
+        raise AttributeError(f'Unknown property {prop.name}')
+
+    def do_set_property(self, prop, value):
+        """Set property callback."""
+
+        self.logger.debug('Setting property "%s" to "%s".', prop.name, value)
+        if prop.name == 'tag':
+            self.tag = value
+        elif prop.name == 'source-id':
+            self.source_id = value
+        else:
+            raise AttributeError(f'Unknown property {prop.name}')
+
+    def handle_buffer(
+        self,
+        sink_pad: Gst.Pad,
+        element: Gst.Element,
+        buffer: Gst.Buffer,
+    ) -> Gst.FlowReturn:
+        """Route buffer either to src_tagged or src_not_tagged pad."""
+
+        self.logger.debug('Handling buffer PTS=%s', buffer.pts)
+
+        not_tagged_buffers = self.parse_buffer(buffer)
+        tagged = not_tagged_buffers is None
+        if tagged != self.last_tagged:
+            self.logger.debug('Switching to tagged=%s', tagged)
+            event = build_stream_part_event(next(self.part_id_gen), tagged)
+            self.src_pad_not_tagged.push_event(event)
+            self.src_pad_tagged.push_event(event)
+
+        self.last_tagged = tagged
+        if tagged:
+            return self.push_buffer(self.src_pad_tagged, buffer)
```

Comment:
Why is there no error logging similar to the one below for the non_tagged_buffers?

Reply:
Added logging into method `push_buffer`.